File: 2DVirtualWorldSimulatorCpp/2DVirtualWorldSimulator/Fox.cpp
#include <iostream>
#include "World.h"
#include "Fox.h"

Fox::Fox() {
	this->name = 'F';
	this->strength = 3;
	this->initiative = 7;
}

void Fox::action(World* world) {
	//wont go to stronger organizm
	int side_to_go = rand() % 4;

	switch (side_to_go) {
	case 0:
		if (position.x - 1 >= 0) {
			dueToStrength(world, position.x - 1, position.y);
		}
		else SetOrganismWantedPosition(position.x, position.y);
		break;
	case 1:
		if (position.x + 1 < world->GetHorizSize()) {
			dueToStrength(world, position.x + 1, position.y);
		}
		else SetOrganismWantedPosition(position.x, position.y);
		break;
	case 2:
		if (position.y - 1 >= 0) {
			dueToStrength(world, position.x, position.y - 1);
		}
		else SetOrganismWantedPosition(position.x, position.y);
		break;
	case 3:
		if (position.y + 1 < world->GetVerticSize()) {
			dueToStrength(world, position.x, position.y + 1);
		}
		else SetOrganismWantedPosition(position.x, position.y);
		break;
	default:
		break;
	}
}

void Fox::dueToStrength(World* world, int posX, int posY) {
	Organism* org = world->getOrganismVectorData(posY, posX);
	if ((org != nullptr && org->GetStrength() < this->GetStrength()) || org == nullptr) {
		SetOrganismWantedPosition(posX, posY);
	}
	else SetOrganismWantedPosition(position.x, position.y);
}

char Fox::GetName() const {
	return 'F';
}

string Fox::GetOrganismFullName() const {
	return "Fox";
}

Fox::~Fox() {

}

File: 2D_sym_python/animals/Fox.py
from Animal import*
import random

class Fox(Animal):

    def __init__(self):
        super().__init__()
        self.name='F'
        self.strength = 3
        self.initiative = 7
    
    def action(self, world):
        side_to_go = random.randint(0,3)

        if side_to_go==0:
            if self.position['x']-1 >= 0:
                self.dueToStrength(world,self.position['x']-1,self.position['y'])
            else: self.SetOrganismWantedPosition(self.position['x'],self.position['y'])
        elif side_to_go == 1:
            if self.position['x']+1 < world.GetHorizSize():
                self.dueToStrength(world,self.position['x']+1,self.position['y'])
            else: self.SetOrganismWantedPosition(self.position['x'],self.position['y'])
        elif side_to_go == 2:
            if self.position['y']-1 >= 0:
                self.dueToStrength(world,self.position['x'],self.position['y']-1)
            else: self.SetOrganismWantedPosition(self.position['x'],self.position['y'])
        elif side_to_go == 3:
            if self.position['y']+1 < world.GetVerticSize():
                self.dueToStrength(world,self.position['x'],self.position['y']+1)
            else: self.SetOrganismWantedPosition(self.position['x'],self.position['y'])
    
    def dueToStrength(self, world, posX, posY):
        org = world.GetOrganismVectorData(posY,posX)
        if (org != None and org.GetStrength() < self.GetStrength()) or org==None:
              self.SetOrganismWantedPosition(posX, posY)
        else: self.SetOrganismWantedPosition(self.position['x'], self.position['y'])
    
    def GetName(self):
        return 'F'
    
    def GetOrganismFullName(self):
        return "Fox"